src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestChangePolicyAction.kt

```diff
@@ -0,0 +1,244 @@
+                }
+            }
+
+            client.search(
+                getManagedIndexConfigSearchQuery(managedIndexUuids.map { (_, indexUuid) -> indexUuid }.toTypedArray()),
+                ActionListener.wrap(::onSearchResponse, ::onFailure)
+            )
+        }
+
+        private fun onSearchResponse(response: SearchResponse) {
+            val foundManagedIndices = mutableSetOf<String>()
+            val sweptConfigs = response.hits
+                    .map {
+                        // The id is the index uuid
+                        foundManagedIndices.add(it.id)
+                        SweptManagedIndexConfig.parseWithType(contentParser(it.sourceRef), it.seqNo, it.primaryTerm)
+                    }
+
+            // If we do not find a matching ManagedIndexConfig for one of the provided managedIndexUuids
+            // it means that we have not yet created a job for that index (which can happen during the small
+            // gap of adding a policy_id to an index and a job being created by the coordinator, or the coordinator
+            // failing to create a job and waiting for the sweep to create the job). We will add these as failed indices
+            // that can not be updated from the ChangePolicy yet.
+            managedIndexUuids.forEach {
+                val (index, indexUuid) = it
+                if (!foundManagedIndices.contains(indexUuid)) {
+                    failedIndices.add(FailedIndex(index, indexUuid, INDEX_NOT_INITIALIZED))
+                }
+            }
+
+            if (sweptConfigs.isEmpty()) {
+                channel.sendResponse(getRestResponse(0, failedIndices))
+            } else {
+                updateManagedIndexConfig(sweptConfigs)
+            }
+        }
+
+        private fun updateManagedIndexConfig(sweptConfigs: List<SweptManagedIndexConfig>) {
+            val mapOfItemIdToIndex = mutableMapOf<Int, Pair<String, String>>()
+            val bulkRequest = BulkRequest()
+            sweptConfigs.forEachIndexed { index, sweptConfig ->
+                bulkRequest.add(updateManagedIndexRequest(sweptConfig.copy(changePolicy = changePolicy)))
+                mapOfItemIdToIndex[index] = sweptConfig.index to sweptConfig.uuid
+            }
+            client.bulk(bulkRequest, onBulkResponse(mapOfItemIdToIndex))
+        }
+
+        private fun onBulkResponse(mapOfItemIdToIndex: Map<Int, Pair<String, String>>): RestResponseListener<BulkResponse> {
+            return object : RestResponseListener<BulkResponse>(channel) {
+                override fun buildResponse(bulkResponse: BulkResponse): RestResponse {
+                    val failedResponses = (bulkResponse.items ?: arrayOf()).filter { it.isFailed }
+                    failedResponses.forEach {
+                        val indexPair = mapOfItemIdToIndex[it.itemId]
+                        if (indexPair != null) {
```

Comment:
mutableMapOf<Int, Pair<String, String>>
If you provide it a correct key, it should give you a Pair<String, String>
If you provide it a key it does not have, it will return null